File: GameHive/Constants/GameTypeEnum.cs
/*************************************************************************************
 * 文 件 名:   GameTypeEnum.cs
 * 描    述: 用于定义支持的游戏类型，并提供中英文名转化
 * 版    本：  V2.0 .NET客户端初版
 * 创 建 者：  Cassifa
 * 创建时间：  2024/11/25 22:04
*************************************************************************************/
namespace GameHive.Constants.GameTypeEnum {
    public enum GameType {
        TicTacToe,
        AntiGo,
        Gobang,
        Gobang88,
        MisereTicTacToe,
    }
    public static class BoardGameExtensions {
        public static string GetChineseName(this GameType game) {
            return game switch {
                GameType.Gobang => "五子棋",
                GameType.Gobang88 => "8*8五子棋",
                GameType.TicTacToe => "井字棋",
                GameType.MisereTicTacToe => "反井字棋",
                GameType.AntiGo => "不围棋",
                _ => "未知"
            };
        }

        public static string GetEnglishName(this GameType game) {
            return game switch {
                GameType.Gobang => "Gobang",
                GameType.Gobang88 => "8*8Gobang",
                GameType.TicTacToe => "Tic-Tac-Toe",
                GameType.MisereTicTacToe => "Misère Tic-Tac-Toe",
                GameType.AntiGo => "ReverseGo",
                _ => "Unknown"
            };
        }
    }
}

File: Backend/gamehive-admin/src/main/java/com/gamehive/constants/GameTypeEnum.java
package com.gamehive.constants;

/**
 * 游戏类型枚举
 * @author Cassifa
 */
public enum GameTypeEnum {
    GOBANG(1, "五子棋", "Gobang"),
    GOBANG_88(2, "8*8五子棋", "8*8Gobang"),
    ANTI_GO(3, "不围棋", "ReverseGo"),
    TIC_TAC_TOE(4, "井字棋", "Tic-Tac-Toe"),
    MISERE_TIC_TAC_TOE(5, "反井字棋", "Misère Tic-Tac-Toe");

    private final int code;
    private final String chineseName;
    private final String englishName;

    GameTypeEnum(int code, String chineseName, String englishName) {
        this.code = code;
        this.chineseName = chineseName;
        this.englishName = englishName;
    }

    public int getCode() {
        return code;
    }

    public String getChineseName() {
        return chineseName;
    }

    public String getEnglishName() {
        return englishName;
    }

    public static GameTypeEnum fromCode(int code) {
        for (GameTypeEnum value : values()) {
            if (value.getCode() == code) {
                return value;
            }
        }
        throw new IllegalArgumentException("未知游戏类型: " + code);
    }
} 